dnf/base.py

```diff
@@ -407,6 +407,7 @@ def reset(self, sack=False, repos=False, goal=False):
                 self._goal = dnf.goal.Goal(self._sack)
             if self._group_persistor is not None:
                 self._group_persistor = self._activate_group_persistor()
+            self._comps_trans = dnf.comps.TransactionBunch()
```

Comment:
shouldn't we check it  for None (as it is done for sack, group persistor and so on)?